src/Nix/Builtins.hs

```diff
@@ -779,8 +787,8 @@ placeHolder = fromValue @Text >=> \_ -> do
 
 absolutePathFromValue :: MonadNix e m => NValue m -> m FilePath
 absolutePathFromValue = \case
-    NVStr pathText _ -> do
-        let path = Text.unpack pathText
+    NVStr ns -> do
+        let path = Text.unpack $ stringIntentionallyDropContext ns
```

Comment:
For now, we should fail if the string has context, but actually in the C++ code base this works through the same mechanism as import-from-derivation: If we do readFile or readDir on a path with conext, we build the derivations we need to build during evaluation and *then* read from the file/directory.

@jwiegley What's the best way to track that we need to change this once we support IFD?

Reply:
Track it with a GitHub issue?